src/main/java/org/miracum/etl/fhirtoomop/listeners/OrganizationStepListener.java

```diff
@@ -0,0 +1,91 @@
+package org.miracum.etl.fhirtoomop.listeners;
+
+import lombok.extern.slf4j.Slf4j;
+import org.miracum.etl.fhirtoomop.DbMappings;
+import org.miracum.etl.fhirtoomop.MemoryLogger;
+import org.miracum.etl.fhirtoomop.repository.OmopRepository;
+import org.springframework.batch.core.ExitStatus;
+import org.springframework.batch.core.StepExecution;
+import org.springframework.batch.core.StepExecutionListener;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.CacheConfig;
+import org.springframework.cache.caffeine.CaffeineCache;
+import org.springframework.cache.caffeine.CaffeineCacheManager;
+import org.springframework.stereotype.Component;
+
+import static org.miracum.etl.fhirtoomop.Constants.VOCABULARY_IPRD;
+import static org.miracum.etl.fhirtoomop.Constants.VOCABULARY_SNOMED;
+import static org.miracum.etl.fhirtoomop.Constants.VOCABULARY_WHO;
+
+@Slf4j
+@Component
+@CacheConfig(cacheManager = "caffeineCacheManager")
+public class OrganizationStepListener implements StepExecutionListener {
+
+    private final OmopRepository repositories;
+    private final DbMappings dbMappings;
+    private static final MemoryLogger memoryLogger = new MemoryLogger();
+    private final Boolean bulkload;
+    @Autowired
+    CaffeineCacheManager cacheManager;
+
+    public OrganizationStepListener(OmopRepository repositories, DbMappings dbMappings, Boolean bulkload) {
+        this.repositories = repositories;
+        this.dbMappings = dbMappings;
+        this.bulkload = bulkload;;
+    }
+
+    @Override
+    public void beforeStep(StepExecution stepExecution) {
+        if (bulkload.equals(Boolean.TRUE)) {
+            log.info("========= Preparing OMOP DB for BulkLoad =========");
+            truncateDb();
+        } else {
+            log.info("========= Preparing OMOP DB for IncrementalLoad =========");
+            cleanUpTable();
+        }
```

Comment:
From my understanding this condition should be added in the first listener of our flow. Please check it and remove if needed.
And i have also seen "========= Preparing OMOP DB for BulkLoad =========" being logged twice in the system.